rotkehlchen/api/rest.py

```diff
@@ -805,6 +806,20 @@ def query_exchange_history_events(
             name: str | None,
     ) -> dict[str, Any]:
         """Queries new history events for the specified exchange and saves them in the database."""
+        if location in (Location.BINANCE, Location.BINANCEUS):
+            exchanges = [
+                x.name for x in self.rotkehlchen.exchange_manager.connected_exchanges.get(location, [])  # noqa: E501
+            ] if name is None else [name]
+            for exchange_name in exchanges:
+                if len(self.rotkehlchen.data.db.get_exchange_credentials_extras(
+                    name=exchange_name,
+                    location=location,
+                ).get(BINANCE_MARKETS_KEY, [])) == 0:
+                    return wrap_in_fail_result(
+                        message=f'Cannot query {exchange_name} exchange events with no markets set.',  # noqa: E501
+                        status_code=HTTPStatus.BAD_REQUEST,
+                    )
```

Comment:
rather than iterating all the excahnges wouldn't it be better to check directly on the db? `set_binance_pairs` might be a good reference. Why I say that is because you avoid iterating the excahnge and do the extra calls on the object